packages/elements/src/dialog/index.ts

```diff
@@ -101,11 +114,38 @@ export class Dialog extends Overlay {
     `];
   }
 
+  private _header: string | null = null;
   /**
-   * Set Header/Title of the dialog
+   * Header element
+   * @param value Header text
    */
   @property({ type: String })
-  public header: string | null = null;
+  public set header (value: string | null) {
+    const oldValue = this._header;
+    const ariaLabelledby = this.getAttribute('aria-labelledby');
+    if(oldValue !== value) {
+      if(this.ariaLabel.length === 0 && ariaLabelledby == null) {
+        this.ariaLabel = String(value);
+      }
+      this._header = value;
+      this.requestUpdate('header', oldValue);
+    }
+  }
+
+  public get header (): string | null {
+    return this._header;
+  }
+
+  /**
+   * Aria indicating aria-label of dialog
+   * @ignore
+   */
+  @property({ type: String,
+    reflect: true,
+    attribute: 'aria-label',
+    converter: { toAttribute: emptyStringToNull } // TODO: Remove after typescript update to allow nullable for ARIAMixin
```

Comment:
Why  empty ? There is always header value in dialog. Also have you look into `aria-labelledby` and link it to `ef-header` and `aria-describedby` to describe dialog ? does it work ?

Reply:
I think that `aria-labelledby` cannot link to `ef-header` which is in the shadow-root.

Reply:
Have you tried it yet ?

Reply:
Yes, I tried it before and it did not work.